### ISSUE TITLE
Requests from CloudFront fail with "KeyError: Content-Type"

### ISSUE THREAD

**Issue by jellybob** [agschwender/pilbox#30]
When requesting an image from our CloudFront bucket we get the backtrace below - requesting the same file direct from its S3 bucket doesn't show this issue. While we could conceivably go direct to S3 every time it would mean rewriting responses from our API layer to do so, which we'd rather not do if its avoidable:

```
[E 150220 12:27:45 web:1407] Uncaught exception GET /?url=https%3A%2F%2Fd1r8772cwfbmfd.cloudfront.net%2Fsystem%2Fproduct_images%2F9404%2Fkl275-big.jpg&w=100&h=100&mode=crop (172.17.42.1)
    HTTPServerRequest(protocol='http', host='hubbub-image-sizer.elasticbeanstalk.com', method='GET', uri='/?url=https%3A%2F%2Fd1r8772cwfbmfd.cloudfront.net%2Fsystem%2Fproduct_images%2F9404%2Fkl275-big.jpg&w=100&h=100&mode=crop', version='HTTP/1.1', remote_ip='172.17.42.1', headers={'X-Real-Ip': '82.71.4.106', 'Host': 'hubbub-image-sizer.elasticbeanstalk.com', 'X-Forwarded-For': '82.71.4.106', 'Accept': '*/*', 'User-Agent': 'curl/7.35.0'})
    Traceback (most recent call last):
      File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1334, in _execute
        result = yield result
      File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 628, in run
        value = future.result()
      File "/usr/local/lib/python2.7/dist-packages/tornado/concurrent.py", line 109, in result
        raise_exc_info(self._exc_info)
      File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 633, in run
        yielded = self.gen.send(value)
      File "/usr/local/lib/python2.7/dist-packages/pilbox/app.py", line 144, in get
        self.render_image(resp)
      File "/usr/local/lib/python2.7/dist-packages/pilbox/app.py", line 194, in render_image
        self._set_headers(resp.headers, outfile_format)
      File "/usr/local/lib/python2.7/dist-packages/pilbox/app.py", line 248, in _set_headers
        self.set_header("Content-Type", headers["Content-Type"])
      File "/usr/local/lib/python2.7/dist-packages/tornado/httputil.py", line 209, in __getitem__
        return dict.__getitem__(self, _normalized_headers[name])
    KeyError: 'Content-Type'
[E 150220 12:27:45 web:1811] 500 GET /?url=https%3A%2F%2Fd1r8772cwfbmfd.cloudfront.net%2Fsystem%2Fproduct_images%2F9404%2Fkl275-big.jpg&w=100&h=100&mode=crop (172.17.42.1) 399.65ms
```

**Comment by agschwender** [comment]
The URL you provided is not returning a Content-Type in its response. It really should be. But I'll put a fix in so that it doesn't cause in error in Pilbox.

**Comment by jellybob** [comment]
Fantastic, thanks. Setting `content_type_from_image` seems to have done the job - to be honest I'd be inclined to make the error clearer, but not automatically do anything, since in most cases its probably a configuration error that there's no Content-Type header.

**Comment by agschwender** [comment]
Yeah, I agree. This is definitely a configuration issue and probably something that should be fixed there. But Pilbox is really designed to return things as its given (even if it's probably not correct). So I think with that in mind, simply not setting the content type and returning the response as given is the "correct" thing to do.

Anyway, this is fixed in d8e471ba23dc8253e876d30b60707e28a7c336e5 (v1.1.5)
